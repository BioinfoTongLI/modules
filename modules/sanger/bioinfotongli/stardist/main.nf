process BIOINFOTONGLI_STARDIST {
    tag "$meta.id"
    label "gpu"
    label 'process_medium'

    // conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'quay.io/cellgeni/tiled_stardist:0.9.1':
        'quay.io/cellgeni/tiled_stardist:0.9.1' }"

    input:
    tuple val(meta), val(x_min),val(y_min),  val(x_max), val(y_max), path(image)

    output:
    tuple val(meta), path("${prefix}_sd_outlines.wkt"), emit: wkts 
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    stardist_helper.py run \\
        --image-path $image \\
        --x_min $x_min \\
        --x_max $x_max \\
        --y_min $y_min \\
        --y_max $y_max \\
        --prefix ${prefix} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(stardist_helper.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_sd_outlines.wkt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(stardist_helper.py version)
    END_VERSIONS
    """
}
