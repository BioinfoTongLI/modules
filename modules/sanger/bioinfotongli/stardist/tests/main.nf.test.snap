{
    "stardist - tif - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_0_0_500_500_sd_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,badd5e6abbb4e33c6ee6bd75bc8612c8"
                ],
                "versions": [
                    "versions.yml:md5,badd5e6abbb4e33c6ee6bd75bc8612c8"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_0_0_500_500_sd_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-11T09:43:53.533814895"
    },
    "stardist - tif": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_0_0_500_500_sd_outlines.wkt:md5,bb69c9f9a25a1ea5cbab1c658e6278e8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,badd5e6abbb4e33c6ee6bd75bc8612c8"
                ],
                "versions": [
                    "versions.yml:md5,badd5e6abbb4e33c6ee6bd75bc8612c8"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_0_0_500_500_sd_outlines.wkt:md5,bb69c9f9a25a1ea5cbab1c658e6278e8"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-11T09:43:21.088659066"
    }
}